### ISSUE TITLE
Secrets backends

### ISSUE THREAD

**Issue by greenape** [Flowminder/FlowKit#2412]
Would it be worthwhile supporting different secrets backends across the components? An obvious example here would be hashicorp vault.

**Comment by jc-harrison** [comment]
By 'different secrets backends', do you mean 'multiple secrets backends, with the possibility to have separate backends for different components', or 'support hashicorp vault (and/or others) as an alternative to docker secrets'?

**Comment by greenape** [comment]
The latter, which I guess encompasses the former.

**Comment by jc-harrison** [comment]
Yes, I don't see why not. Do you have a specific situation in mind where this would be useful?

**Comment by greenape** [comment]
Well, any scenario where docker swarm isn’t the deploy method really. Or where it is, and you want something more robust for credential rotation for example.
